### ISSUE TITLE
Broken format specifier for sprintf

### ISSUE THREAD

**Issue by gsmwhatnot** [Ai-Thinker-Open/GPRS_C_SDK#172]
## 1. SDK version（SDK 版本）

{

2.018

}

---

## 2. In what kind of operation problems appear, and how to reproduce the problem ?（什么样的操作步骤问题会出现，是否是稳定复现，如何复现问题？）

{

While using sprintf command I can't use some format specifiers. Here are some examples : 

`sscanf("AT+CUSD=1,\"*2*1#\",15\r", "AT+CUSD=%d,%[^,],%d\r",&ussdOption, ussdString, &ussdDcs);`
Output should be like this : 

`ussdOption = 1`
`ussdString = "*2*1#"`  // with double quotations
`ussdDcs = 15`

But it is not ! `Trace` function shows absolute garbage for ussdString.

This one works fine though : 
`sscanf("AT+CUSD=1 \"*2*1#\" 15\r", "AT+CUSD=%d %s %d\r",&ussdOption, ussdString, &ussdDcs);` // replaced COMMA with 'SPACE'
Output will be like this : 

`ussdOption = 1`
`ussdString = "*2*1#"`  // with double quotations
`ussdDcs = 15`

Simply put, it is not possible to format at least a simple comma delimited string.

}

---

**Comment by Neutree** [comment]
sccanf not support regular expressions
